diff --git a/src/spatial/modules/main/spatial_functions_table.cpp b/src/spatial/modules/main/spatial_functions_table.cpp
index 1cc61ad..5d8c307 100644
--- a/src/spatial/modules/main/spatial_functions_table.cpp
+++ b/src/spatial/modules/main/spatial_functions_table.cpp
@@ -133,7 +133,7 @@ struct ST_GeneratePoints {
 		set.AddFunction(generate_points);
 
 		// Overload with seed
-		generate_points.GetArguments() = {GeoTypes::BOX_2D(), LogicalType::BIGINT, LogicalType::BIGINT};
+		generate_points.SetArgumentTypes({GeoTypes::BOX_2D(), LogicalType::BIGINT, LogicalType::BIGINT});
 		set.AddFunction(generate_points);
 		loader.RegisterFunction(set);
 
